### PR TITLE
process data in models

### PR DESCRIPTION
**Pull request by saifullah767**
In this PR I have implemented the following

### Create models
-  Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
-  Set up associations between models.

### Use models to insert data
- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user by running the following code:
    ```ruby
       first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
       second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
    ```
- Create at least 4 posts written by one of the users you created by running the following code:
    ```ruby
       first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
    ```
- Create at least 6 posts comments for one of the posts you created by running the following code:
    ```ruby
       Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
    ```
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

### Create custom methods
Your models should include some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.
- Go to `rails c` and check if your methods are working.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  # Method that updates the posts counter for a user.
+
+  def update_posts_counter
+    author.update(post_counter: author.posts.count)
+  end
```
Comment (CollinsTatang) [inline app/models/post.rb]: **Nice job creating the different counter update method correctly**
### Additional Support

- [x] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private

- [x] Please do the same for `after_save :update_comments_counter` and `after_save :update_likes_counter`

```ruby
# app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  # Method which returns the 5 most recent comments for a given post.
+
+  def recent_comments
+    comments.last(5)
+  end
```
Comment (CollinsTatang) [inline app/models/post.rb]: **Great** job so far implementing the project requirement

- [ ] **Required** This method returns the 5 most recent comments for a given post not the last 5 most recent comments.
**You Can Use This Solution**
```ruby
def recent_comments
  comments.order('created_at Desc').limit(5)
end
```

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @saifullah767 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by saifullah767** [comment]
# Applied the required changes.

image

image

**Review by peterrobert — APPROVED** [review]
Hi @saifullah767, 

### Points to note:

1. Great work implementing all the changes.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**